pyxem/utils/calibration_utils.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2022 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import numpy as np
+
+from diffsims.generators.library_generator import DiffractionLibraryGenerator
+from pyxem.utils.indexation_utils import index_dataset_with_template_rotation
+
+
+def find_diffraction_calibration(
+    patterns,
+    calibration_guess,
+    library_phases,
+    lib_gen,
+    size,
+    max_excitation_error=0.01,
+    **kwargs
+):
+    """Finds the diffraction calibration for a pattern or set of patterns by maximizing correlation scores.
+
+    Parameters
+    ----------
+    patterns : hyperspy.signals.Signal2D
+        Diffration patterns to be iteratively matched to find maximum correlation scores.
+    calibration_guess : float
+        Inital value for the diffraction calibration in inverse Angstoms per pixel
+    library_phases : diffsims.libraries.StructureLibrary
+        Dictionary of structures and associated orientations for which
+        electron diffraction is to be simulated.
+    lib_gen : diffsims.generators.DiffractionLibraryGenerator
+        Computes a library of electron diffraction patterns for specified atomic
+        structures and orientations.  Used to create the DiffractionLibrary.
+    size : integer
+        How many different steps to test for the first two iterations. These steps have a size of 1% of the calibration guess.
+    max_excitation_error : float
+        Maximum exacitation error.  Default is 0.01.
+    kwargs
+        Keyword arguments passed to :meth:`index_dataset_with_template_rotation`.
+    Returns
```

Comment:
Blank line before Returns

Reply:
Same goes for the other two functions.

Reply:
Question on a "minimal working example":  should it be simulated data only then or would it be better if I include a small dataset to use with it?

Reply:
Since the functionality is meant to be used with a few patterns, adding a small data set to go with the notebook should be OK, right? If it's easy to do, I suggest to do that. Simulated would be fine as well, I think.